data/registry_model/registry_oci_model.py

```diff
@@ -825,6 +837,51 @@ def is_namespace_enabled(self, namespace_name):
         namespace = model.user.get_namespace_user(namespace_name)
         return namespace is not None and namespace.enabled
 
+    def lookup_cached_manifest_by_digest(
+        self,
+        model_cache,
+        repository_ref,
+        manifest_digest,
+        allow_dead=False,
+        allow_hidden=False,
+        require_available=False,
+        raise_on_error=False,
+    ):
+        def load_manifest():
+            manifest = self.lookup_manifest_by_digest(
+                repository_ref,
+                manifest_digest,
+                allow_dead,
+                allow_hidden,
+                require_available,
+                raise_on_error,
+            )
+
+            if manifest:
+                manifest_dict = manifest.asdict()
+                manifest_dict["internal_manifest_bytes"] = manifest_dict[
+                    "internal_manifest_bytes"
+                ].as_unicode()
+                manifest_dict["inputs"]["repository"] = manifest_dict["inputs"][
+                    "repository"
+                ].asdict()
+                manifest_dict["inputs"]["legacy_image_handler"] = None  # TODO(kleesc): Remove
+                manifest_dict["inputs"]["legacy_id_handler"] = None  # TODO(kleesc): Remove
```

Comment:
did you miss removing the above 2 lines?

Reply:
These should be removed once https://github.com/quay/quay/pull/2555 is merged. For now, they're still expected.